packages/hardhat-core/test/internal/hardhat-network/provider/utils/assertEqualTraces.ts

```diff
@@ -14,18 +14,51 @@ export function assertEqualTraces(
 
   assert.equal(actual.structLogs.length, expected.structLogs.length);
 
-  for (const [i, log] of expected.structLogs.entries()) {
+  for (const [i, expectedLog] of expected.structLogs.entries()) {
+    const actualLog = actual.structLogs[i];
     // we ignore the gasCost of the last step because
     // we don't guarantee that it's correct
     if (i === expected.structLogs.length - 1) {
-      actual.structLogs[i].gasCost = 0;
-      log.gasCost = 0;
+      actualLog.gasCost = 0;
```

Comment:
I don't remember this code, but I see that we were previously also modifying the expected result.

I'm not sure this is a good idea, as we're introducing a side-effect to a function that's merely supposed to be asserting equality.

Instead, I believe that we should either be modifying Hardhat to comply with the EIP or accepting a wider range of values for properties that we know EDR doesn't support yet (like `STATICCALL` gas costs), similar to what Franco pointed out in our Slack discussion

Reply:
Good call on not modifying the input. I fixed it by making copies in 373e99b4b73aa8b7b6013e10f20aa474533b6d51. 

This is testing against fixtures, so the alternative would be to modify the fixtures which is very cumbersome.